### DISCUSSION TITLE
Tag suggestions

### DISCUSSION THREAD

**Discussion by jb3** [python-discord/meta#86] [General]
# :warning: Thread discontinued, please read [here](https://github.com/python-discord/meta/discussions/86#discussioncomment-666726).

Have a suggestion for a new tag? Comment your ideas below, make sure to include a bit of overview on what the tag should mention!

We're looking for tags that are primarily Python related, or at least community related. If you aren't sure feel free to comment it and we'll let you know if it fits the tag system.

Additionally, make sure to look through others tag suggestions and upvote any you like!

In order to add a tag, you will need to add a file in the `bot/resources/tags` folder of our [`bot`](git.pydis.com/bot) repository, in a markdown format. [This link](https://github.com/python-discord/bot/new/main/bot/resources/tags) will drop you directly in the right folder!

**Comment by Akarys42** [comment]
For the record, we had quite a few suggestion on https://github.com/python-discord/bot/issues/810 that are still not implemented, they could be worth considering.

**Reply by jb3** [reply]
Good point, yeah. I've locked that issue now and redirected stuff here, but anyone wanting to work on tags should by all means check out that list. :rocket:

**Comment by Xithrius** [comment]
I was thinking about creating a `lambda` tag to explain anonymous functions.

Some things to consider in the tag would be the following:
- What is an anonymous function? Why should I care?
- When to use a lambda function.
- When to define a function.
- Lambda as a callable argument vs. `functools.partial` (maybe, still thinking about this one).

**Reply by janine9vn** [reply]
I would be super down for a lambda vs partial thing, maybe as a separate tag. It's something I have code snippets set aside just to explain what a user might want to use when.

**Reply by Xithrius** [reply]
In the end we'd probably could have `lambda`, `partial` and `lambda vs partial`.

**Comment by anand2312** [comment]
I'd like to suggest a tag explaining KeyErrors, and the different options available to fix them (`dict.get`, Defaultdicts)

**Reply by Akarys42** [reply]
Sounds good to me!

**Reply by Den4200** [reply]
PR: python-discord/bot#1398

**Reply by jb3** [reply]
Added!

**Comment by MrKomodoDragon** [comment]
I’d like to suggest a tag for blocking/non-blocking, (e.g. the right modules/libraries to use, and why using blocking tools is bad sometimes)

**Reply by janine9vn** [reply]
A tag is usually a short snippet to get information across to quickly explain a specific problem or pitfall. This feels very general and might be better suited for an article (which I think would be great to have).

**Reply by Akarys42** [reply]
I agree with kutie, this seems like a topic better suited to an article. We can write a tag referring to the article later if needs be.

**Reply by MrKomodoDragon** [reply]
Maybe a short thing with blocking/nonblocking? That’s what I meant. Not an entire tag on concurrency 😆

**Reply by jb3** [reply]
I think a brief explainer on how to do things in parallel with Python with some pointers to libraries and other articles (maybe even a guide of our own if we make one) is a nice idea.

**Reply by MrKomodoDragon** [reply]
Ok

**Comment by Brodevil** [comment]
I am suggesting a tab name pip.
This will be defiantly beneficial for all the beginners to understand the pip in detail and start to make it in use.
It will well be explained and easy to understand.

**Reply by Brodevil** [reply]
Please tell me about this idea
What's the other suggestion 
Thank you!

**Reply by jb3** [reply]
I think there's a lot to explain about pip and maybe a tag for it isn't the best place. We could look into writing some sort of article/guide on it but I think that if we do go down the route of writing about pip a tag is not the place for it due to all the platform differences and so on.

**Reply by Brodevil** [reply]
Ok sir, no issue

**Comment by Urkchar** [comment]
Add a tag for when a user leaks their token or their bot's token in an image. If they leak it in text, the Python bot will respond. If they leak it in an image, the Python bot will not tell them what to do.
![image](https://user-images.githubusercontent.com/49205496/107858039-993a0c00-6df7-11eb-8e25-d27bda785e6d.png)

**Comment by anand2312** [comment]
I think a tag on the difference between classes and their instances would be quite useful.

**Comment by anand2312** [comment]
What do you guys think about a tag on strftime? A short explanation along with a few of the common format codes.

**Reply by Xithrius** [reply]
I'd say go for it. Hopefully it won't be too long. Be sure to include [this website](<https://strftime.org/>).

**Comment by MrKomodoDragon** [comment]
A tag for the humanize library and its features?

**Comment by jb3** [comment]
We'll be discontinuing this thread in favour of individual issues opened on the ~[python-discord/bot](https://github.com/python-discord/bot)~ [python-discord/meta](https://github.com/python-discord/meta) repository.

If you would like to implement a tag please leave a comment on the relevant issue there (if you would like to create one from the pool above please transfer it into an issue first), after that we'll assign you and you can PR as necessary.

We've made changes to make it easier to contribute tags, PRs only changing tags now require only one approval from a staff member.
